src/main/kotlin/com/terraformation/backend/tracking/db/Queries.kt

```diff
@@ -26,9 +33,98 @@ fun latestObservationForSubzoneField(observationIdField: Field<ObservationId?>)
               // Prefer results from the same observation as the `observationId` if it exists.
               // True is considered greater than false, so sorting by an "=" expression in
               // descending order means the matches come first.
-              OBSERVATION_ID.eq(observationIdField).desc(),
+              OBSERVATIONS.ID.eq(observationIdField).desc(),
               // Otherwise take the results from the next latest observation for that area.
               OBSERVATIONS.COMPLETED_TIME.desc(),
           )
           .limit(1)
     }
+
+fun plotIsInObservationResult(
+    monitoringPlotIdField: Field<MonitoringPlotId?>,
+    observationIdField: Field<ObservationId?>,
+    isPermanent: Boolean,
+): Condition {
```

Comment:
If preferred I can work on simplifying this to a single query (I recognize that it's technically a single query now, just split into variables). Many of the subqueries are only used once. I had them split out as I was writing them because I was trying to grok the query while writing it

Reply:
I think keeping it split up will make it easier to understand in the future.